### DISCUSSION TITLE
Cell voltage statistics MQTT/InfluxDB

### DISCUSSION THREAD

**Discussion by mr-manuel** [Louisvdw/dbus-serialbattery#376] [Q&A] [answered]
Hello,

I'm looking for a way to store all cell voltages in InfluxDB. My idea was to send the different voltages from the driver as JSON via MQTT to Node-RED, where I can create a flow to store the data in the time series database.

Has someone done this already? Is there maybe another way to get the single cell voltages without adding MQTT to the driver?

I'm using the Venus OS Large Image with Node-RED.

**Comment by Louisvdw** [comment, marked as answer]
All the data that the driver publish to the dbus should already be available in MQTT in Node Red.
See https://github.com/victronenergy/dbus-mqtt/blob/master/README.md if you have not yet enabled MQTT on your GX and how to read data.

The cell voltages are published at 
battery.{port}/Voltages/Cell#
![image](https://user-images.githubusercontent.com/26077665/209933204-488e4d8d-40df-41ed-8e3e-e2b182dd193b.png)

**Comment by mr-manuel** [comment]
Indeed it is. It works also for multiple batteries. Thanks!

`N/{VRM_PORTAL_ID}/battery/{1...}/Voltages/#`

![grafik](https://user-images.githubusercontent.com/12592765/209938037-3647ed1e-aabb-4c9d-a10b-e5f02263924a.png)

**Reply by Besenreiter** [reply]
Hello, on my system the CellVoltages are not published via MQTT.
RPI3 with VenusOS (FirmewareBuild: {"value": "20221016213031"}
BMS is a Daly 250A with 16s. When I connect with PC to the DALY, I can see the cellvoltages. So the Daly obviously is putting them on the interface. I connect the same interface to the Venus, so I think the celvoltages should get to the Venus. But the MQTT-Broker doesn't send them out. Every Topic wich could contain something like cellvoltages (N/b827eb9432fe/battery/1/Cell/1 or N/b827eb9432fe/battery/1/Balances/Cell1 or N/b827eb9432fe/battery/1/Voltages/Cell1) contain "null". Anybody any idea what goes wrong on my system?
Thanks!
![grafik](https://user-images.githubusercontent.com/18124405/216819076-aa0f48eb-36da-439d-a1f6-6f21575f2ac0.png)

**Reply by mr-manuel** [reply]
Do you see the cell voltages in the Venus OS GUI? You can also check via ssh and `dbus-spy` like in the screenshot of Luis above

**Reply by Besenreiter** [reply]
Thanks for your Answer. And, no in the Venus OS Gui I can only see:
![grafik](https://user-images.githubusercontent.com/18124405/216828435-047eb84d-e172-4fd5-ad12-03d3dc760e8c.png)
Not the cell voltages itself

**Reply by Besenreiter** [reply]
dbus-spy. I didn't know that. realy cool tool. Butr there are'nt the voltages as well:
 
![grafik](https://user-images.githubusercontent.com/18124405/216828625-4cbffabf-f3f2-446e-919d-921414a6ac7a.png)

**Reply by Besenreiter** [reply]
![grafik](https://user-images.githubusercontent.com/18124405/216828646-57a936db-3eb3-451f-b1ca-95fd9b3fbeb2.png)

**Reply by mr-manuel** [reply]
As you see the driver is not providing the cell voltages. Maybe @Louisvdw can help you.

**Reply by Louisvdw** [reply]
It does seem that some Daly BMS does not publish the cell voltages the same as their protocol states. I have another report from a Daly BMS that also does not publish them. 

The only real way would be to sit with an example and test what is happening. And I don't have one 😢

**Reply by Besenreiter** [reply]
But the BMS does publish the voltages, as I can see them on the same Interface when I disconnect it from the Pi (Venus) and connect to a PC and use "DalyBmsMonitorV9". There you can see the voltages. Maybe the dataformat is somehow different to the usual one?
If that would help, I can try to sniff the serial traffic and send you the log. Would that help?
